x/interchaintxs/keeper/ibc_handlers.go

```diff
@@ -1,17 +1,61 @@
 package keeper
 
 import (
+	"strings"
 	"time"
 
 	"github.com/cosmos/cosmos-sdk/telemetry"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
 
-	"github.com/neutron-org/neutron/internal/sudo"
+	contractmanagertypes "github.com/neutron-org/neutron/x/contractmanager/types"
 	"github.com/neutron-org/neutron/x/interchaintxs/types"
 )
 
+const (
+	// GasReserve is the amount of gas on the context gas meter we need to reserve in order to add contract failure to keeper
```

Comment:
lines 16-57 the same in the transfer wrapper and interchain txs, is it possible to reuse the common code?

Reply:
This is the problem, I want to have modules to be independent as possible. In order to reuse common code we need to introduce new entity, that will be used between different modules, but I do not want to break this modularity.

Reply:
yup, we discussed it with Albert; let's leave it as is